### ISSUE TITLE
Scale the number of dask workers to available resources on HPC

### ISSUE THREAD

**Issue by MarieNyst** [dask/dask#6483]
When using dask on a HPC, it would be useful to be able to automatically scale up or down the number of dask workers based on how much free resources there are.

A simple use case would be:
I launch my computations, there are 20 free nodes. Dask sees that and starts 100 workers to get those nodes allocated.
1 hour later, a colleague launches something that is now in the queue. Dask sees that and each time a worker finishes a computation it terminates it until the queued jobs start.
A few hours later, my colleague gets their results and dask starts new worker to claim unused resources.

I thought that it was what the [adapt](https://docs.dask.org/en/latest/setup/adaptive.html) function was all about. Then understood that it doesn't adapt to the cluster scheduler load but to dask scheduler load.

Now, most of my researches point to this adapt method. Either because what I'm asking doesn't exist or because it is not so much used...

**Comment by mrocklin** [comment]
I agree that this would be valuable.  I think that the main challenge is figuring out how to ask the underlying hardware how much free capacity it has.  I'm not sure that there is a generic way to do this.

I'm going to move this over to the dask-jobqueue repository because I think that people there will be more informed on this topic.
